sn_interface/src/types/cache/mod.rs

```diff
@@ -138,18 +133,18 @@ where
         }
 
         for key in expired_keys {
-            let _prev = self.items.write().await.remove(&key);
+            let _prev = self.items.remove(&key);
         }
     }
 
     /// Remove items that exceed capacity, oldest first.
-    async fn drop_excess(&self) {
-        let len = self.len().await;
+    fn drop_excess(&mut self) {
+        let len = self.len();
         if len > self.capacity {
             let excess = len - self.capacity;
             let excess_keys: Vec<_>;
             {
-                let read_items = self.items.read().await;
+                let read_items = &self.items;
                 let mut items = read_items.iter().collect_vec();
```

Comment:
No need for a let binding for the reference anymore.
```suggestion
                let mut items = self.items.iter().collect_vec();
```